_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+depending on the size of the work,
+we may create several sub-issues under a single high-level issue
+so that progress reports aren't "I'm working on it" several days in a row.
+
+-   Summary: Biweekly prioritization to determine team focus for the sprint
+-   Key Questions:
+    -   What is the next slice of work we want to deliver?
+    -   What small improvements should we prioritize?
+    -   What key bugs should we prioritize?
+    -   What tech debt needs to be addressed?
+-   Frequency: biweekly
+-   Responsible: Scrum master and technical leads
+-   Contributors: engineering team members
+-   Products:
+    -   Labeled issues
+
+> I've worked on teams where issues had a "percent complete" field
+> that the people responsible for it were supposed to update daily.
+> It wasn't helpful:
+> it's easy to see how much ditch you have left to dig,
+> but much harder to know how much code you have left to write and debug.
+> Breaking milestones down into inch-pebbles
+> and marking the latter as "finished" or "not finished"
+> has proven much more useful.
+
+### Daily Standups
+
+We don't actually stand up,
+but this is still the most important meeting each team has.
+The tech lead of the project does a quick lap around the team to find out
+what each person has done,
+what they're planning to do,
+and what's holding them back.
+We try to enforce a three-sentence rule:
+if a discussion during individual updates goes on longer than that
+we defer it until after everyone has finished their initial report.
+
+The "output" from standups is course corrections.
+Should we try another approach to a problem?
+Did someone discover an issue that ought to go in the backlog for the next sprint,
+or something that's important enough to replace something planned for this sprint?
+
+-   Summary: Daily team sync
+-   Key Questions:
+    -   What did you work on yesterday?
+    -   What are you working on today?
+    -   What is blocking you?
+-   Frequency: daily
+-   Responsible: technical leads
+-   Contributors: engineering team members
+-   Products:
+    -   Updated issues
+
+### Sprint Retros
+
+Just as the whole of engineering does a retrospective each quarter,
+each team does a short retrospective at the end of each sprint.
+Sometimes the output is "let's try doing something differently"
+(for a smaller value of "something" than would be discussed by the whole team in a quarterly retro);
+other times,
+the output is just,
+"Well done!"
+because it's as important to celebrate victories as it is to learn from defeats.
+
+-   Summary: Bi-weekly sync to reflect on the past sprint and identify possible improvements
+-   Key Questions:
+    -   What went well?
+    -   What didn't go well?
+    -   What should we improve/change/adopt to make things better?
+-   Frequency: biweekly
+-   Responsible: Scrum master
+-   Contributors: engineering team members
+-   Products:
+    -   Updated issues and processes
```

Comment:
Question: In the above retro section you had the following products
    -   Ranked list of things that worked well and things that didn't
    -   Ranked list of ideas for things to try to adopt, improve, or drop

Should the products be more or less the same? Or is there a reason why you wanted to differentiate them?

Reply:
good catch